lua/telescope/config.lua

```diff
@@ -34,131 +58,273 @@ function config.set_pickers(pickers)
+))
+
+-- A table of all the usual defaults for telescope.
+-- Keys will be the name of the default,
+-- values will be a list where:
+-- - first entry is the value
+-- - second entry is the description of the option
+
+local telescope_defaults = {
+
+  sorting_strategy = {
+    "descending",
+    [[
     Determines the direction "better" results are sorted towards.
 
     Available options are:
     - "descending" (default)
-    - "ascending"]])
+    - "ascending"]],
+  },
 
-  set("selection_strategy", "reset", [[
+  selection_strategy = {
+    "reset",
+    [[
     Determines how the cursor acts after each sort iteration.
 
     Available options are:
     - "reset" (default)
     - "follow"
-    - "row"]])
+    - "row"]],
+  },
 
-  set("scroll_strategy", "cycle", [[
+  scroll_strategy = {
+    "cycle",
+    [[
     Determines what happens you try to scroll past view of the picker.
 
     Available options are:
     - "cycle" (default)
-    - "limit"]])
+    - "limit"]],
+  },
+
+  layout_strategy = {
+    "horizontal",
+    [[
+    Determines the default layout of Telescope pickers.
+    See |telescope.layout| for details of the available strategies.
 
-  set("layout_strategy", "horizontal")
-  set("layout_defaults", {})
+    Default: 'horizontal']],
+  },
 
-  set("width", 0.75)
-  set("winblend", 0)
-  set("prompt_position", "bottom")
-  set("preview_cutoff", 120)
+  layout_config = { layout_config_defaults, layout_config_description },
 
-  set("results_height", 1)
-  set("results_width", 0.8)
+  winblend = { 0 },
 
-  set("prompt_prefix", "> ", [[
+  prompt_prefix = { "> ", [[
     Will be shown in front of the prompt.
 
-    Default: '> ']])
-  set("selection_caret", "> ", [[
+    Default: '> ']]
+  },
+
+  selection_caret = { "> ", [[
     Will be shown in front of the selection.
 
-    Default: '> ']])
-  set("entry_prefix", "  ", [[
+    Default: '> ']]
+  },
+
+  entry_prefix = { "  ", [[
     Prefix in front of each result entry. Current selection not included.
 
-    Default: '  ']])
-  set("initial_mode", "insert")
+    Default: '  ']]
+  },
+
+  initial_mode = { "insert" },
 
-  set("border", {})
-  set("borderchars", { '─', '│', '─', '│', '╭', '╮', '╯', '╰'})
+  border = { {} },
 
-  set("get_status_text", function(self)
-    local xx = (self.stats.processed or 0) - (self.stats.filtered or 0)
-    local yy = self.stats.processed or 0
-    if xx == 0 and yy == 0 then return "" end
+  borderchars = { { "─", "│", "─", "│", "╭", "╮", "╯", "╰" } },
 
-    return string.format("%s / %s", xx, yy)
-  end)
+  get_status_text = {
+    function(self)
+      local xx = (self.stats.processed or 0) - (self.stats.filtered or 0)
+      local yy = self.stats.processed or 0
+      if xx == 0 and yy == 0 then
+        return ""
+      end
+
+      return string.format("%s / %s", xx, yy)
+    end,
+  },
+
+  dynamic_preview_title = {
+    false,
+    [[
+Will change the title of the preview window dynamically, where it
```

Comment:
Some of the descriptions are not indented (I am not the biggest fan of starting at col 1 here) :smile: